index.js

```diff
@@ -68,6 +68,15 @@ module.exports = {
   included: function() {
     this._super.included.apply(this, arguments);
 
+    let isApp = !this.project.isEmberCLIAddon();
```

Comment:
We won't need to detect app/addon, since for addon development `ember-fetch` should lives lin `devDependencies` it's also top-level.
You may intend to force error for Addon + dependencies.

Reply:
I think we need to detect if the project is an app or not. I want this error to be thrown during the application build only.

 Also, ember-fetch can be a transitive-dependency for add-on if the add-on has a dependency on another add-on which is dependent on ember-fetch. In that case it should not error. The top-level dependency requirement is only meant for application. 

Let me know if I missed anything here.